### ISSUE TITLE
Error executing stored procedure - Procedure or function has too many arguments specified.

### ISSUE THREAD

**Issue by KeishaW** [MightyOrm/Mighty#20]
I get the error: _Procedure or function [stored_proc_name] has too many arguments specified_, when using the example for executing a stored procedure.
```
var result = db.ExecuteProcedure("my_add_proc",
    inParams: new { a = 1, b = 2}, outParams: new { c = (int?)null }); // error thrown
Console.WriteLine(result.c); 
```
However, when I removed the `outParams` section it was succesfully exceuted. 
```
var result = db.ExecuteProcedure("my_add_proc",
    inParams: new { a = 1, b = 2}); // works and execute but with empty ResultView in result variable
```

I confirmed by executing the stored procedure in SQL Server Management Studio, that it does, in fact, return a value, however running it using the example returns the error.

Even though removing the `outParams` section allows it to execute, I would still like to access the returned value.

**Comment by mikebeaton** [comment]
TL/DR: You need to use `returnParams` not `outParams`.

Stored procedures in `System.Data.Common` (and therefore Mighty) allow input, output, input-output and return params. In the case of SQL Server, output and input-output types can *both* be used on SP params marked `OUTPUT`, and return is used for the return value of the SP.

So you need to use `returnParams` not `outParams` to get the return value. You might wonder why it's called `returnParams`, plural, when in fact there can only ever be one SP return value in SQL Server. But nothing in `System.Data.Common` - and so nothing in Mighty - enforces this. I think if you added multiple return parameters they'd all just get set to the return value - but I'm not quite sure! If working against SQL Server, you should only add one return param, which should be `int` or `int?` to be sure of operating in a supported manner. (You're already doing that, as soon as you change to `returnParams`.)
